main/windows/index.ts

```diff
@@ -234,18 +261,22 @@ class Tray {
     return this.ready
   }
 
-  setContextMenu (menu: Menu) {
-    return this.electronTray.setContextMenu(menu)
+  isVisible () {
+    return (windows.tray as BrowserWindow).isVisible()
+  }
+
+  canAutoHide () {
+    return !this.recentElectronTrayClick && store('main.autohide') && !store('windows.dash.showing') && !frameManager.isFrameShowing()
   }
 
   hide (autohide: boolean = false) {
     store.toggleDash('hide');
     if (autohide) {
-        this.recentAutohide = true;
-        clearTimeout(this.recentAutoHideTimeout as NodeJS.Timeout);
-        this.recentAutoHideTimeout = setTimeout(() => {
-            this.recentAutohide = false;
-        }, 50);
+      this.recentAutohide = true;
+      clearTimeout(this.recentAutoHideTimeout as NodeJS.Timeout);
```

Comment:
semicolons

Reply:
when the formatting PR finally goes in I will die happy